### PR TITLE
fix: upsert table before upserting datasource node

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

- table needs to be upserted first, as the upsert datasource node looks for an existing table 
- the word "table" is reserved and needs quoting

### CODE REVIEW

File: core/src/stores/postgres.rs [hunk 1]
```diff
@@ -2912,7 +2912,7 @@ impl Store for PostgresStore {
         let stmt = c
             .prepare(
                 "INSERT INTO data_sources_nodes \
-               (id, data_source, created, node_id, timestamp, title, mime_type, parents, document, table, folder) \
+               (id, data_source, created, node_id, timestamp, title, mime_type, parents, document, \"table\", folder) \
```
Comment (tdraier) [inline core/src/stores/postgres.rs:2915]: we also have the same in the select in get_data_source_node

Comment (fontanierh) [reply core/src/stores/postgres.rs:2915]: fixed thanks

**Review by tdraier — APPROVED** [review]

File: core/bin/dust_api.rs [hunk 2]
```diff
@@ -2091,29 +2092,43 @@ async fn tables_upsert(
...
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "internal_server_error",
+                "Failed to upsert table",
+                Some(e),
+            )
+        }
+    };
+
+    // Upsert the data source node if title and mime_type are present
+    if let Some(n) = &maybe_ds_node {
```
Comment (spolu) [inline core/bin/dust_api.rs:2110]: Eventually (when title and mimeType are required) we might want to run this from the `upsert_table` function directly right?

Comment (fontanierh) [reply core/bin/dust_api.rs:2110]: Maybe but now obvious. I think for now the paradigm is that store functions manage one specific table.

If we do it like this then we should probably remove entirely the store functions to manage data source nodes directly (and only expose functions to manage tables, docs and folders)

I don't really have an opinion on this yet

Comment (spolu) [reply core/bin/dust_api.rs:2110]: We could add an upsert function on Table that calls the 2 separate store functions but using a shared transaction?

Comment (spolu) [reply core/bin/dust_api.rs:2110]: Not a big deal anyway. It's just that we are open to having some tables without nodes if the node upsert fails here...

Comment (fontanierh) [reply core/bin/dust_api.rs:2110]: Yeah we can def have that it's a good idea (even though technically we only have a single place where we upsert tables so we could pontially share the Tx in the handler).

Still I think having that on `Table` is a cleaner

Comment (spolu) [reply core/bin/dust_api.rs:2110]: Only caveat is that in front we're moving towards the invariant that transactions should only exist in resources... store.rs is our cheap resource equivalent here so we're going against the direction we're taking in front....

food for thoughts

**Pull request merged**